packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -26,27 +26,27 @@ interface EmberArray<T> extends Enumerable {
   ): NativeArray<U>;
   mapBy<K extends string>(key: K): NativeArray<Value<T, K>>;
   filter<Target>(
-    callback: (this: Target, item: T, index: number, arr: this) => boolean,
+    callback: (this: Target, item: T, index: number, arr: this) => unknown,
```

Comment:
I believe this is correct. We don't want to force people to return a specific type, allowing people to pass values that will get evaluated as truthy/falsy.